src/repositories/request-repository.ts

```diff
@@ -386,6 +386,18 @@ export class RequestRepository {
     return query
   }
 
+  /**
+   * Returns the number of pending anchor requests that remain in the database.
+   * @returns The number of requests in the database in status PENDING
+   */
+  public async countPendingRequests(): Promise<number> {
```

Comment:
should we put a note that this is a db call and should not be made too frequently?  Or we assume people know that

Reply:
This whole RequestRepository class exists solely to contain the interactions with the request table in the database, so I think it's sufficiently implied